src/lib/components/ComboBox.svelte

```diff
@@ -0,0 +1,250 @@
+<script generics="K" lang="ts">
+	import { createEventDispatcher } from 'svelte';
+	import type { ChangeEventHandler } from 'svelte/elements';
+	const dispath = createEventDispatcher<{
+		selectOption: {
+			value: string;
+			display: string;
+		};
+	}>();
+
+	type Options = [K[], (arg: K) => string, (arg: K) => string | number];
```

Comment:
can we name the arguments in this tuple

Reply:
or use a json object i dont know why i made the decision to use a tuple. though for sake of simple refactoring just named tuple